### ISSUE TITLE
MacOs failure - The remote certificate is invalid because of errors in the certificate chain: NotTimeValid

### ISSUE THREAD

**Issue by chcosta** [dotnet/dnceng#1485]
Investigation needed for https://github.com/dotnet/runtime/issues/91705

Error message (on MacOS):
```
       System.Net.Http.HttpRequestException : The SSL connection could not be established, see inner exception.
      ---- System.Security.Authentication.AuthenticationException : The remote certificate is invalid because of errors in the certificate chain: NotTimeValid
```

Recent example: https://[helixre107v0xd1eu3ibi6ka.blob.core.windows.net/dotnet-runtime-refs-heads-release-60-staging-ef87e00e954c42c39b/System.Net.Http.Functional.Tests/1/console.502a9bda.log?helixlogtype=result](https://helixre107v0xd1eu3ibi6ka.blob.core.windows.net/dotnet-runtime-refs-heads-release-60-staging-ef87e00e954c42c39b/System.Net.Http.Functional.Tests/1/console.502a9bda.log?helixlogtype=result)

A couple of questions:
- Is this failure consistent or intermittent?
  - A quick search shows it's common for an artifact to be signed with a valid certificate which then expires and this causes a "NotTimeValid" error from System.Security.Cryptography.X509.  Often, users ignore the error, but even then (on MacOS), the failure still pops.  See https://github.com/dotnet/runtime/issues/38005.
- It's plausible that this is only occurring in dotnet/runtime (and is related to the above), in which case, I'm not sure that this is something we can (should) drive to resolution. Are we seeing this failure in other repos?  If this is occurring in multiple repos, then an ICM may be required.

FYI @ilyas1974 , @lewing 

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements
### Release Note Description

**Comment by chcosta** [comment]
Looking into this now

**Comment by chcosta** [comment]
I don't believe this is an infrastructure issue, see https://github.com/dotnet/runtime/issues/91705#issuecomment-1830464861

**Comment by lewing** [comment]
Thank you for taking a look at this
